### ISSUE TITLE
Adaptive neural network optimized for solving specific issue

### ISSUE THREAD

**Issue by mxrguspxrt** [tensorflow/tensorflow#1767]
Hey!

From Kolmogorov theorem we can accept that neural network is able to represent any function, but it is unknown what should be the specific configuration of the network.

TensorFlow has a lot of helpful parts (behind scenes back prop, learning rate decay, gradient optimizers etc), but the topology of the network (how many layers and neurons) is currently arbitrary and the best solution is found by manual testing.

Would it make sense to have auto of the box some kind of  "find the best solution for my classification problem"? Just adding layers, neurons and autoencoders to find the best solution? 

Genetic programming of neural network. Plugin & play black box would be awesome? :)

**Comment by martinwicke** [comment]
A limited version of this is called hyperparameter optimization. You can check out projects like spearmint (https://github.com/JasperSnoek/spearmint). In general, it's a far off research problem. 

I'm closing this issue since it's not a well-defined feature request. Better support for hyperparameter tuning is planned.
